### PR TITLE
Updates installer instructions for Mac OS

### PR DESCRIPTION
**Pull request by SomtochiAma**

### CODE REVIEW

**Comment by atoato88** [comment]
I tried `make` on Mac, and `make` success to build binary.
But its binary is Linux exec binary, not Mac exec binary.
The reason why you created PR is this?
If so, IMHO, how about update `build` target in `Makefile` to build Mac exec binary if run on Mac?

@justinsb @stealthybox 
Any ideas?

**Comment by SomtochiAma** [comment]
> But its binary is Linux exec binary, not Mac exec binary.
> The reason why you created PR is this?
Yes, that's the reason
> If so, IMHO, how about update `build` target in `Makefile` to build Mac exec binary if run on Mac?
I could raise a PR for that

File: installer/Makefile [hunk 1]
```diff
@@ -53,7 +54,13 @@ vendor:
 	go mod verify
 
 only-build:
+ifeq ($(UNAME),Darwin)
+	echo Darwin
+	make binary
```
Comment (atoato88) [inline installer/Makefile:59]: In Mac, `make binary` is called and it calls `autogen` and `vendor` prerequisites, on the other hand, In Linux, this target calls `$(MAKE) shell ...` which doesn't call  `autogen` and `vendor` prereqiusites.
So, this is not same action, I think.

How about change [`shell` target](https://github.com/kubernetes-sigs/cluster-addons/blob/af88a3421fa5edcad9666e9a8eefe91298cb3612/installer/Makefile#L12-L23) to add `GOOS` and `GOARCH` on `docker run` in Mac?
When specify `GOOS=darwin` and `GOARCH=amd64`, go compiles binary for Mac.

**Comment by atoato88** [comment]
/lgtm
Thank you for update.

**Pull request closed without merging**